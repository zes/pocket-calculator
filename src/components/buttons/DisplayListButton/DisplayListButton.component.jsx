import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import GenericButton from '../GenericButton';

export default class DisplayListButton extends PureComponent {
  render() {
    const { toggleDisplayList } = this.props;
    return (
      <GenericButton
        action={() => toggleDisplayList()}
        className="listbutton"
        value="LIST"
      />
    );
  }
}

DisplayListButton.propTypes = {
  toggleDisplayList: PropTypes.func.isRequired,
};
