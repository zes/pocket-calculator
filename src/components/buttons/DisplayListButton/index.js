export { default } from './DisplayListButton.container';
