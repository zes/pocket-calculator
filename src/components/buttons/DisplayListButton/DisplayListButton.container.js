import { connect } from 'react-redux';
import { toggleDisplayList } from '../../../redux/actions';

import DisplayListButton from './DisplayListButton.component';

const mapDispatchToProps = {
  toggleDisplayList,
};

export default connect(
  null,
  mapDispatchToProps,
)(DisplayListButton);
