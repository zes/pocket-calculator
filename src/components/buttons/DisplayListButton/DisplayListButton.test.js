import React from 'react';
import renderer from 'react-test-renderer';
import DisplayListButton from './DisplayListButton.component';

describe('[Component] <DisplayListButton />', () => {
  const props = {
    toggleDisplayList: jest.fn(),
  };

  it('renders a snapshot', () => {
    const tree = renderer.create(<DisplayListButton {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
