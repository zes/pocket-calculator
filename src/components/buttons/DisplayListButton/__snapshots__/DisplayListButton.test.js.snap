// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] <DisplayListButton /> renders a snapshot 1`] = `
<button
  className="button listbutton"
  onClick={[Function]}
  type="button"
>
  LIST
</button>
`;
